src/core/expr/fexpr_count_countna.cc

```diff
@@ -99,37 +99,37 @@ class FExpr_CountNA : public FExpr_Func {
         case SType::VOID:
         case SType::BOOL:
         case SType::INT8:
-          return make<int8_t, int64_t>(std::move(col), gby, is_grouped);
+          return make<int8_t>(std::move(col), gby, is_grouped);
         case SType::INT16:
-          return make<int16_t, int64_t>(std::move(col), gby, is_grouped);
+          return make<int16_t>(std::move(col), gby, is_grouped);
         case SType::DATE32: 
         case SType::INT32:
-          return make<int32_t, int64_t>(std::move(col), gby, is_grouped);
+          return make<int32_t>(std::move(col), gby, is_grouped);
         case SType::TIME64: 
         case SType::INT64:
-          return make<int64_t, int64_t>(std::move(col), gby, is_grouped);
+          return make<int64_t>(std::move(col), gby, is_grouped);
         case SType::FLOAT32:
-          return make<float, int64_t>(std::move(col), gby, is_grouped);
+          return make<float>(std::move(col), gby, is_grouped);
         case SType::FLOAT64:
-          return make<double, int64_t>(std::move(col), gby, is_grouped);
+          return make<double>(std::move(col), gby, is_grouped);
         case SType::STR32:
         case SType::STR64:
-          return make<CString, int64_t>(std::move(col), gby, is_grouped);                
+          return make<CString>(std::move(col), gby, is_grouped);                
         default:
           throw TypeError()
             << "Invalid column of type `" << stype << "` in " << repr();
       }
     }
 
 
-    template <typename T, typename U>
+    template <typename T>
     Column make(Column &&col, const Groupby& gby, bool is_grouped) const {
       if (is_grouped) {
-        return Column(new Latent_ColumnImpl(new Count_ColumnImpl<T, U, COUNTNA, true>(
+        return Column(new Latent_ColumnImpl(new Count_ColumnImpl<T, int64_t, COUNTNA, true>(
           std::move(col), gby
         )));
       } else {
-        return Column(new Latent_ColumnImpl(new Count_ColumnImpl<T, U, COUNTNA, false>(
+        return Column(new Latent_ColumnImpl(new Count_ColumnImpl<T, int64_t, COUNTNA, false>(
```

Comment:
I don't think we need to pass it to `Count_ColumnImpl` either, because it is always `int64_t` and you know this inside the class already.

Reply:
@oleksiyskononenko it is used to generate the column stype in reduce_unary: 
```py
template <typename T, typename U, bool IS_GROUPED>
class ReduceUnary_ColumnImpl : public Virtual_ColumnImpl {
  protected:
    Column col_;
    Groupby gby_;
    

  public:
    ReduceUnary_ColumnImpl(Column &&col, const Groupby& gby)
      : Virtual_ColumnImpl(gby.size(), stype_from<U>),
        col_(std::move(col)),
        gby_(gby)
    {
      xassert(col_.can_be_read_as<T>());
    }


    ColumnImpl *clone() const override {
      return new ReduceUnary_ColumnImpl(Column(col_), Groupby(gby_));
    }


    size_t n_children() const noexcept override {
      return 1;
    }


    const Column &child(size_t i) const override {
      xassert(i == 0);
      (void)i;
      return col_;
    }
};
```
open to a better way of doing this

Reply:
Yes, but for that you don't need to pass `int64_t` to `Count_ColumnImpl`, because this class already knows it is going to have `int64` stype and `int64_t` element type.